taf/repositoriesdb.py

```diff
@@ -54,37 +55,43 @@ def load_repositories(auth_repo, repo_classes=None, factory=None,
   elif commit in _repositories_dict[auth_repo.name]:
     return
 
-  if commit is None:
-    commit = auth_repo.head_commit_sha()
+  if not len(commits):
+    commits = [auth_repo.head_commit_sha()]
   if root_dir is None:
     root_dir = Path(auth_repo.repo_path).parent
 
-  repositories_dict = {}
-  _repositories_dict[auth_repo.name][commit] = repositories_dict
+  for commit in commits:
+    repositories_dict = {}
+    # check if already loaded
+    if commit in _repositories_dict[auth_repo.name]:
+      continue
 
-  targets = auth_repo.get_json(commit, targets_path)
-  repositories = auth_repo.get_json(commit, repositories_path)
+    _repositories_dict[auth_repo.name][commit] = repositories_dict
 
-  # target repositories are defined in both mirrors.json and targets.json
-  repositories = repositories['repositories']
-  targets = targets['signed']['targets']
-  for path, repo_data in repositories.items():
-    urls = repo_data['urls']
-    target = targets.get(path)
-    if target is None and only_load_targets:
-      continue
-    additional_info = _get_custom_data(repo_data, targets[path])
+    repositories = _get_json_file(auth_repo, repositories_path, commit)
+    targets = _get_json_file(auth_repo, targets_path, commit)
+
+    # target repositories are defined in both mirrors.json and targets.json
+    repositories = repositories['repositories']
+    targets = targets['signed']['targets']
+    for path, repo_data in repositories.items():
+      urls = repo_data['urls']
+      target = targets.get(path)
+      if target is None and only_load_targets:
+        continue
+      additional_info = _get_custom_data(repo_data, targets[path])
 
-    if factory is not None:
-      git_repo = factory(root_dir, path, urls, additional_info)
-    else:
-      git_repo_class = _determine_repo_class(repo_classes, path)
-      git_repo = git_repo_class(root_dir, path, urls, additional_info)
+      if factory is not None:
+        git_repo = factory(root_dir, path, urls, additional_info)
+      else:
+        git_repo_class = _determine_repo_class(repo_classes, path)
+        git_repo = git_repo_class(root_dir, path, urls, additional_info)
 
-    if not isinstance(git_repo, GitRepository):
-      raise Exception(f'{type(git_repo)} is not a subclass of GitRepository')
+      if not isinstance(git_repo, GitRepository):
+        raise Exception(f'{type(git_repo)} is not a subclass of GitRepository')
 
-    repositories_dict[path] = git_repo
+      repositories_dict[path] = git_repo
+  get_deduplicated_repositories(auth_repo, commits[0])
 
 
 def _determine_repo_class(repo_classes, path):
```

Comment:
I know this is not changed, but I missed it in the last CR. This should work the same:

```python
def _determine_repo_class(repo_classes, path='default'):
  # if no class is specified, return the default one
  if repo_classes is None:
    return GitRepository
  # if only one value is specified, that means that all target repositories
  # should be of the same class
  if not isinstance(repo_classes, dict):
    return repo_classes

  return repo_classes.get(path, GitRepository)
```